crates/ruff_linter/src/rules/flake8_pyi/rules/string_or_bytes_too_long.rs

```diff
@@ -49,6 +49,16 @@ pub(crate) fn string_or_bytes_too_long(checker: &mut Checker, string: StringLike
         return;
     }
 
+    if let Some(depr_message) =
+        warnings_dot_deprecated_message(checker.semantic().current_expression_parent(), checker)
+    {
+        if let StringLike::StringLiteral(literal) = string {
+            if depr_message == literal {
+                return;
```

Comment:
I think for simplicity I might suggest structuring this like `is_docstring_stmt` -- so just return `true` if we're in one of those decorators, rather than comparing the literal.

Reply:
Yeah, I wondered about that. It would make it simpler, but could possibly lead to false negatives if people are (incorrectly) passing in string literals as a second or third argument to `@deprecated`? Only the first argument should really be excluded. But I'll admit it's unlikely that people would pass in string literals as a second or third argument of `@deprecated`, since it only has one non-keyword-only argument at runtime. So maybe the simpler solution is better here. I'll defer to your judgement :)

Reply:
Hmm I see. I'd be okay with the simpler method, but if we retain this, I think it needs to be tweaked a bit, since `StringLike` can be an f-string or a bytes literal. Instead of comparing based on type and contents, can we just compare the ranges to see if the first argument is `string`?

Reply:
Yeah, I changed my mind -- have amended the PR to go with the simpler approach :)

I think I was worrying to much -- declining to emit this error on any arguments inside a call to `@warnings.deprecated` should be fine. False negatives are obviously less bad than false positives in general, but that feels especially true here: the whole premise of PYI053 is basically a somewhat fuzzy heuristic for spotting some questionably long default values in stubs.

Plus, it feels really unlikely that anybody would trigger this hypothetical false negative in practice (since you'd have to write code that type checkers would flag, and runtime Python would reject). _And_ flake8-pyi also just ignores all string literals inside calls to `@warnings.deprecated`, so we'll be consistent with the original Python implementation of this rule :)

Reply:
I definitely could've gone either way here -- it seemed "simple enough" to me that it was a better complexity tradeoff to just go with this approach, but I can also see how it would be subtly wrong in some ways.